kingpin/actors/aws/ecs.py

```diff
@@ -0,0 +1,242 @@
+        'cluster': (str, REQUIRED,
+                    'ECS cluster in the region to launch the task into'),
+        'count': (int, 1, 'How many tasks to run'),
+        'skip_wait': (bool, False,
+                      'Whether to skip waiting for the tasks to complete')
+    }
+
+    @gen.coroutine
+    def _register_task(self, task_definition):
+        """Registers a task on ECS.
+
+        Args:
+            task_definition: dict of ECS task definition parameters.
+
+        Returns:
+            tuple: ('family', 'revision').
+        """
+        if self._dry:
+            self.log.warning('Would register task')
+            return task_definition['family'], '1'
+
+        response = self.ecs_conn.register_task_definition(**task_definition)
+
+        # Parse data from the server's response.
+        task_definition = response['taskDefinition']
+        family = task_definition['family']
+        revision = task_definition['revision']
+
+        self.log.info('Task definition {0}:{1} registered'.format(
+            family, revision))
+        return family, revision
+
+    @gen.coroutine
+    @dry('Would run task {0}:{1}')
+    def _run_task(self, family, revision):
+        """Runs a task on ECS given a task definition's family and revision.
+
+        Args:
+            family: Task family string
+            revision: Task revision string
+
+        Returns:
+            list: task ARNs.
+        """
+        response = self.ecs_conn.run_task(
+            cluster=self.option('cluster'),
+            taskDefinition='{family}:{revision}'.format(
+                family=family,
+                revision=revision),
+            count=self.option('count'))
+        self._handle_failures(response['failures'])
+        return [t['taskArn'] for t in response['tasks']]
+
+    @gen.coroutine
+    @dry('Would wait for tasks')
+    def _wait_for_tasks(self, tasks):
+        """Wait for tasks to stop on ECS.
+
+        If a task stops with a non-zero exit code,
+        this will raise an exception.
+
+        Args:
+            tasks: list of task ARNs to wait for.
+        """
+        while True:
+            response = self.ecs_conn.describe_tasks(
+                cluster=self.option('cluster'),
+                tasks=tasks
+            )
+
+            self._handle_failures(response['failures'])
+
+            # List of containers per task.
+            # Tasks can run on multiple containers.
+            containers_for_tasks = [t['containers'] for t in response['tasks']]
+            # Compute the total count of running tasks.
+            total_count = sum(len(x) for x in containers_for_tasks)
+            stopped_count = 0
+
+            for containers in containers_for_tasks:
+                for task in containers:
+                    if task['lastStatus'] == 'STOPPED':
+                        stopped_count += 1
+                    exit_code = task.get('exitCode', None)
```

Comment:
again, synchronous api call... only pointing this one out because loops and async calls get interesting in tornado. You're going to want to re-factor this a bit. @Mikhail I'm sure will help out a little bit here. :)